File: flask_app/app/services/user_tier_metrics_service.py
from app.dao.user_tier_metrics_dao import UserTierMetricsDAO

class UserTierMetricsService:

    @staticmethod
    def get_for_user_month(user_id, month_year):
        return UserTierMetricsDAO.get_for_user_month(user_id, month_year)

    @staticmethod
    def create_or_update(data):
        return UserTierMetricsDAO.create_or_update(data)

    @staticmethod
    def get_all_metrics():
        return UserTierMetricsDAO.get_all_metrics()

    @staticmethod
    def get_top_spenders(month_year, top_n=10):
        return UserTierMetricsDAO.get_top_spenders(month_year, top_n)

File: java_membership_api/src/main/java/com/example/membership_api/service/UserTierMetricsService.java
package com.example.membership_api.service;

import com.example.membership_api.dao.UserTierMetricsDAO;
import com.example.membership_api.model.UserTierMetrics;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserTierMetricsService {

    @Autowired
    private UserTierMetricsDAO dao;

    public UserTierMetrics getForUserMonth(String userId, String monthYear) {
        return dao.getForUserMonth(userId, monthYear);
    }

    public UserTierMetrics createOrUpdate(String userId, int orderCount, double totalSpent) {
        return dao.createOrUpdate(userId, orderCount, totalSpent);
    }

    public List<UserTierMetrics> getAllMetrics() {
        return dao.getAllMetrics();
    }

    public List<UserTierMetrics> getTopSpenders(String monthYear, int topN) {
        return dao.getTopSpenders(monthYear, topN);
    }
}
